news/templates/news/user_news.html

```diff
@@ -1,47 +1,48 @@
-    <h4 class="panel-title">{% trans "Course news" %}</h4>
-  </div>
-  <div class="list-group">
-    {% for item in news %}
-    <a class="list-group-item{% if item.collapsed %} collapsed{% endif %}" href="#news{{ item.id }}" data-toggle="collapse" role="button" aria-controls="news{{ item.id }}" aria-expanded="{% if item.collapsed %}false{% else %}true{% endif %}">
-      <h4 class="list-group-item-heading">
-        <span class="caret"></span>
-        {{ item.title|safe }} <small>{{ item.publish }}</small>
-        {% if not item|is_published:now %}<small><span class="label label-default">{% trans "Appears later" %}</span></small>{% endif %}
-        {% if is_course_staff %}<small><span class="label label-default">{{ item.audience|news_audience }}</span></small>{% endif %}
-      </h4>
-    </a>
-    <div class="list-group-item-text collapse{% if not item.collapsed %} in{% endif %}" id="news{{ item.id }}">
-      {{ item.body|safe }}
-    </div>
-    {% if item.begin_more %}
-    <a class="list-group-item" href="#more-news" data-toggle="collapse" role="button" aria-controls="more-news" aria-expanded="false">
-      {% trans "Show more" %}
-    </a>
-    <div class="collapse" id="more-news">
-    {% endif %}
-    {% endfor %}
-    {% if more > 0 and news|length > more %}
-    </div>
-    {% endif %}
-  </div>
-</div>
+	<div class="panel panel-default news-panel">
+		<div class="panel-heading">
+			<h3 class="panel-title">{% trans "Course news" %}</h3>
+		</div>
+		<div class="list-group">
+			{% for item in news %}
+				<div class="{% if item.pin %}pinned-{% endif %}list-group-item">
+					{% if item.pin %}
+						<span class="sr-only">{% trans "Pinned news item on top of the list" %}</span>
+					{% endif %}
+					<div class="list-group-item-heading">
+						<h4 class="list-group-item-title">
+							{{ item.title|safe }}
+						</h4>
+						<div class="list-group-item-details">
+							{% if not item|is_published:now %}
+								<span class="future-instance">{% trans "Published on" %}: <time>{{ item.publish }}</time></span>
+							{% else %}
+								<time class="current-instance">{{ item.publish }}</time>
+							{% endif %}
+							{% if is_course_staff %}
+								<span class="label label-primary">{{ item.audience|news_audience }}</span>
+							{% endif %}
+							{% if item.pin %}
+								<span class="glyphicon glyphicon-pushpin" aria-hidden="true"></span>
+							{% endif %}
+						</div>
+					</div>
+					<p class="list-group-item-text">
+						{{ item.body|safe }}
+					</p>
+				</div>
+				{% if forloop.counter == more and news|length > more %}
+					<a class="folding-list-group-item collapsed" href="#more-news" data-toggle="collapse" role="button" aria-controls="more-news" aria-expanded="false">
+						{% trans "Show older" %} <span class="caret"></span>
+					</a>
+					<div class="collapse" id="more-news">{# start of div.collapse for news #}
+				{% endif %}
+			{% endfor %}
+			</div>{# end of div.collapse for news #}
```

Comment:
When adding of `<div>` is condiftional, then `</div>` has to be too or otherwise the html layout breaks. This is basically the reason why these should be two different lists so things like this wouldn't happen. @jaguarfi's screenshot should be result of this.

Reply:
Weird that it did not come up earlier, since Matthew also did his screenreader test after that change and neither of us noticed anything. I wonder if I made a mistake somewhere yesterday making those screen reader -fixes. Let me check and go back in time a bit. Sorry for the extra hassle.

Reply:
So looks like there was some error earlier in the simplifying commit https://github.com/apluslms/a-plus/pull/570/commits/cb16f33f109e176150f16461f98b67c277c4fc68. 
I am not sure what happened, why Matthew or I did not spot it, maybe my testing-course did not have enough other content (and long enough toc for it to skip on the right-hand-side). But never the less, of course it needs to be fixed.
 
Here in the current master there is the conditional div which I mentioned earlier, that also breaks the html if there are exactly three news items. https://github.com/apluslms/a-plus/blob/d34f94b7c9a8e4d22cd28d08968b36787bb344c4/news/templates/news/user_news.html#L35

This is how three news items looks with the current master:

![image](https://user-images.githubusercontent.com/43036333/86209738-e1affc80-bb7b-11ea-861b-b734b8c17f8d.png)

What do you think, should I do revert to time before that simplifying commit and implement the two separate lists then? even though like @raphendyr said that they are slow? Personally I also think that for readibility that might be better. 

Or should I remove that https://github.com/apluslms/a-plus/commit/cb16f33f109e176150f16461f98b67c277c4fc68 -commit and just try to find a work around for the situation where there are exactly three items?

Reply:
I think the first if requires just `forloop.counter == more` and second div should be fine with `news|length > more`. Just need to check that the index go right, but I think it should.

Reply:
That did not actually quite do it. Since if the "more" was set to 0, as is the case for instance with the "last visited content"-index option, no folding is present. Also had to take care that the ending-div wasn't printed with each round but only for the final item. But thanks for the tip, it gave me another idea :) 

The fix has now been implemented in the final commit https://github.com/apluslms/a-plus/pull/570/commits/f2eada6ca0039192cda99e754e9daf2ccabe2076, it is probably not the prettiest, but I commented and tested it and at least it fixes the three-item-thing at the same time:

**More than three items**
![image](https://user-images.githubusercontent.com/43036333/86239515-25206000-bba8-11ea-82b1-5871fa2f076f.png)

**And three items**
![image](https://user-images.githubusercontent.com/43036333/86239809-abd53d00-bba8-11ea-9811-3c09b0590796.png)